source/fundamentals/promises.txt

```diff
@@ -0,0 +1,194 @@
+They are often used to execute operations that take some time and have
+when you want to notify once it's completed. For more information on Promises
+and related terminology, see the MDN documentation on
+:mdn:`Promises <Web/JavaScript/Reference/Global_Objects/Promise>`.
+
+If you call a driver method such as ``findOneAndModify()``,
+``countDocuments()``, or ``update()``, the NodeJS driver returns a Promise.
+
+You can use the ``await`` operator inside of an ``async`` method on the
+Promise to cause the enclosing method to pause until the Promise reaches
+either the **Fulfilled** or **Rejected** state. If it reaches the **Fulfilled**
+state, the reference contains the operation result as an object or value.
+
+.. code-block:: js
+
+   async function run() {
+     ...
+     const result = await collection.insertOne({ name: "Mount Everest", meters: 8848 });
+
+     // print the result of the insertOne() call
+     console.log(result);
+   }
+
+If you want to execute logic once the Promise reaches the **Fulfilled**
+or **Rejected** state, you can append the ``then()`` method to the Promise.
+The first parameter is the method that gets called when the Promise reaches
+the **Fulfilled** state and the optional second parameter is the method that
+gets called when it reaches the **Rejected** state. These methods are called
+**callbacks** because they execute only after the Promise completes.
+
+.. code-block:: js
+
+   ...
+   collection.updateOne({ name: "Mount McKinley" }, { meters: 6190 })
+     .then(
+       res => console.log(`Updated ${res.result.n} documents`),
+       err => console.error(`Something went wrong: ${err}`)
+     );
+
+If you only need to handle transitions to the **Rejected** state, use the
+``catch()`` method which accepts a single callback, executed when the Promise
+transitions to the **Rejected** state.
+
+.. code-block:: js
+
+     const deleteResult = await collection.deleteOne({ name: "Mount Doom" })
+       .catch(err => console.error(`Fatal error occurred: ${err}`));
+
+.. note::
+
+   Certain methods in the driver such as ``find()`` return a ``Cursor``
+   instead of a Promise. To determine what type each method returns, refer to
+   the :node-api:`NodeJS API documentation <>`.
+
+Callbacks
+---------
+
+A callback is a method that gets called after another method has
+finished executing. This allows the enclosing method to continue to execute
+other commands until the original operation completes. Callbacks are often
+used to enforce the order of processing commands.
+
+In the MongoDB NodeJS driver, you can optionally declare a callback method to
+async operations that normally return Promises such as ``findOneAndUpdate()``,
+``countDocuments()``, or ``update()``. Once the operation completes execution,
+the callback method executes as shown in the following code snippet:
+
+.. code-block:: js
+
+   collection.findOneAndUpdate({ name: "Barronette Peak" }, { name: "Baronette Peak" },
```

Comment:
An update operator is required here.

Reply:
Oops, you mean options object right? https://mongodb.github.io/node-mongodb-native/3.5/api/Collection.html#findOneAndUpdate

Reply:
Never mind! Need both